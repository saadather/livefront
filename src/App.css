.container {
  max-width: 800px;
  margin: 0 auto;
}

.heading {
  text-align: center;
  margin-bottom: 20px;
  text-transform: capitalize;
}

.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.card {
  width: 48%;
  margin-bottom: 20px;
  border: 1px solid #ddd;
  border-radius: 8px;
  overflow: hidden;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  transition: transform 0.2s ease-in-out;
}

.card:hover {
  transform: scale(1.05);
}

.card-image {
  width: 100%;
  height: auto;
  border-bottom: 1px solid #ddd;
}

.card-content {
  padding: 15px;
}

.card-title {
  font-size: 1.2rem;
  margin-bottom: 10px;
}

.card-abstract {
  margin-bottom: 15px;
}

.card-info {
  font-size: 0.9rem;
  margin-bottom: 5px;
}

.read-more {
  display: block;
  background-color: #007bff;
  color: #fff;
  padding: 8px 12px;
  text-decoration: none;
  border-radius: 5px;
  transition: background-color 0.2s ease-in-out;
}

.read-more:hover {
  background-color: #0056b3;
}

.navbar {
  display: flex;
  justify-content: space-around;
  margin-bottom: 20px;
}

.active-link {
  color: #007bff;
}
