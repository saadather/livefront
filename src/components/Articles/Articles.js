import React, { useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';

function Articles({ category }) {
  const [articles, setArticles] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    const fetchArticles = async () => {
      setLoading(true);
      try {
        const response = await fetch(`https://api.nytimes.com/svc/mostpopular/v2/${category}/7.json?api-key=1eHFZtK4jQRIkZeIoIw4ki2agUueFSV8`);
        const data = await response.json();
        setArticles(data.results);
      } catch (error) {
        setError(error);
      } finally {
        setLoading(false);
      }
    };

    fetchArticles();
  }, [category]);

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return <div>Error: {error.message}</div>;
  }

  if (articles.length === 0) {
    return <div>No articles found.</div>;
  }
  
  return (
    <div className="container">
    <h1 className="heading">New York Times Articles</h1>
    <nav className="navbar">
      <NavLink to="/most-popular/emailed" className={({ isActive }) =>
        isActive ? 'active-link' : ''
      }>
        Most Emailed
      </NavLink>
      <NavLink to="/most-popular/shared" className={({ isActive }) =>
        isActive ? 'active-link' : ''
      }>
        Most Shared
      </NavLink>
      <NavLink to="/most-popular/viewed" className={({ isActive }) =>
        isActive ? 'active-link' : ''
      }>
        Most Viewed
      </NavLink>
    </nav>
      <h2 className="heading">{category} Articles</h2>
      <div className="card-container">
        {articles.map(article => (
          <div key={article.id} className="card">
            
            <div className="card-content">
              <h3 className="card-title">{article.title}</h3>
              <p className="card-abstract">{article.abstract}</p>
              <p className="card-info">Published Date: {article.published_date}</p>
              <p className="card-info">Section: {article.section}</p>
              <a href={article.url} className="read-more" target="_blank" rel="noopener noreferrer">
                Read more
              </a>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
}

export default Articles;
