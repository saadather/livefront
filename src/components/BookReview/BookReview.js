// BookReview.js
import React, { useState } from 'react';

const BookCard = ({ book }) => (
  <div className="book-card">
    <h3>{book.book_title}</h3>
    <h4>{book.book_author}</h4>
    <p>{book.publication_dt}</p>
    <p>{book.summary}</p>
  </div>
);

const BookReview = () => {
  const [searchQuery, setSearchQuery] = useState('');
  const [reviews, setReviews] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);

  const handleSearch = async () => {
    try {
      setLoading(true);
      setError(null);

      const response = await fetch(`https://api.nytimes.com/svc/books/v3/reviews.json?author=${searchQuery}&api-key=1eHFZtK4jQRIkZeIoIw4ki2agUueFSV8`);
      const data = await response.json();

      setReviews(data.results);
    } catch (error) {
      setError(error);
    } finally {
      setLoading(false);
    }
  };

  return (
    <div>
      <h2>Book Reviews</h2>
      <div>
        <input
          type="text"
          placeholder="Enter book title"
          value={searchQuery}
          onChange={(e) => setSearchQuery(e.target.value)}
        />
        <button onClick={handleSearch}>Search</button>
      </div>

      {loading && <p>Loading...</p>}
      {error && <p>Error: {error.message}</p>}

      {reviews.length > 0 && (
        <div>
          <h3>Search Results</h3>
          <ul>
            {reviews.map((book) => (
              <BookCard key={book.display_title} book={book} />
            ))}
          </ul>
        </div>
      )}
    </div>
  );
};

export default BookReview;
