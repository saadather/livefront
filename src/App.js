import React from 'react';
import { BrowserRouter as Router, Route, NavLink, Routes } from 'react-router-dom';
import './App.css';
import Articles from './components/Articles/Articles';
import BookReview from './components/BookReview/BookReview';

function App() {
  return (
    <Router>
      <div className="container">
        <nav className="left-nav">
          <NavLink to="/most-popular" className={({ isActive }) =>
              isActive ? 'active-link' : ''
            }>
            Most Popular
          </NavLink>
          <NavLink to="/book-review" className={({ isActive }) =>
              isActive ? 'active-link' : ''
            }>
            Book Review
          </NavLink>
        </nav>
        <div className="main-content">
          <Routes>
            <Route path="/most-popular" element={<Articles category="emailed" />} />
            <Route path="/book-review" element={<BookReview />} />
            <Route path="most-popular/emailed" element={<Articles category="emailed" />} />
            <Route path="most-popular/shared" element={<Articles category="shared" />} />
            <Route path="most-popular/viewed" element={<Articles category="viewed" />} />
          </Routes>
        </div>
      </div>
    </Router>
  );
}

export default App;