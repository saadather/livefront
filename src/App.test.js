import React from 'react';
import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from './App';
import Articles from './components/Articles/Articles';
import BookReview from './components/BookReview/BookReview';

// Mock the fetch function
jest.mock('node-fetch');

describe('App Component', () => {
  it('renders without crashing', async () => {
    render(<App />);
    await waitFor(() => {
      expect(screen.getByText(/Most Popular/)).toBeInTheDocument();
    });
  });

  it('matches snapshot', async () => {
    const { container } = render(<App />);
    await expect(container).toMatchSnapshot();
  });
});

describe('Articles Component', () => {
  // Mock the API response
  const mockApiResponse = {
    status: 'OK',
    results: [
      {
        "uri": "nyt://article/5be53850-75ca-5b13-a5fa-c3016b602e77",
        "url": "https://www.nytimes.com/2024/01/19/science/cicadas-emergence-broods.html",
        "id": 100000009253628,
        "asset_id": 100000009253628,
        "source": "New York Times",
        "published_date": "2024-01-19",
        "updated": "2024-01-22 23:11:44",
        "section": "Science",
        "subsection": "",
        "nytdsection": "science",
        "adx_keywords": "Forests and Forestry;Cicadas (Insects);Insects;Mount St Joseph University;Smithsonian Institution;Alabama;Arkansas;Georgia;Illinois;Indiana;Iowa;Kentucky;Louisiana;Michigan;Midwestern States (US);Mississippi;Missouri;North Carolina;South Carolina;Southeastern States (US);Tennessee;United States;Virginia;Wisconsin;Southern States (US)",
        "column": null,
        "byline": "By Aimee Ortiz",
        "type": "Article",
        "title": "The World Hasnâ€™t Seen Cicadas Like This Since 1803",
        "abstract": "Brood XIX and Brood XIII will both emerge this spring. The last time these bugs showed up at the same time in the United States, Thomas Jefferson was president.",
        "des_facet": [
          "Forests and Forestry",
          "Cicadas (Insects)",
          "Insects"
        ],
        "org_facet": [
          "Mount St Joseph University",
          "Smithsonian Institution"
        ],
        "per_facet": [],
        "geo_facet": [
          "Alabama",
          "Arkansas",
          "Georgia",
          "Illinois",
          "Indiana",
          "Iowa",
          "Kentucky",
          "Louisiana",
          "Michigan",
          "Midwestern States (US)",
          "Mississippi",
          "Missouri",
          "North Carolina",
          "South Carolina",
          "Southeastern States (US)",
          "Tennessee",
          "United States",
          "Virginia",
          "Wisconsin",
          "Southern States (US)"
        ],
        "media": [
          {
            "type": "image",
            "subtype": "photo",
            "caption": "Two cicada groups, known as broods, are set to appear this spring in a dual emergence that will span across the Midwest and Southeastern United States.",
            "copyright": "Suzanne DeChillo/The New York Times",
            "approved_for_syndication": 1,
            "media-metadata": [
              {
                "url": "https://static01.nyt.com/images/2024/01/17/multimedia/00xp-cicada-1-btmz/00xp-cicada-1-btmz-thumbStandard.jpg",
                "format": "Standard Thumbnail",
                "height": 75,
                "width": 75
              },
              {
                "url": "https://static01.nyt.com/images/2024/01/17/multimedia/00xp-cicada-1-btmz/00xp-cicada-1-btmz-mediumThreeByTwo210.jpg",
                "format": "mediumThreeByTwo210",
                "height": 140,
                "width": 210
              },
              {
                "url": "https://static01.nyt.com/images/2024/01/17/multimedia/00xp-cicada-1-btmz/00xp-cicada-1-btmz-mediumThreeByTwo440.jpg",
                "format": "mediumThreeByTwo440",
                "height": 293,
                "width": 440
              }
            ]
          }
        ],
        "eta_id": 0
      }
    ],
  };

  beforeEach(() => {
    // Mock the fetch function to return a Promise with the mocked API response
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockApiResponse),
      })
    );
  });

  it('renders without crashing', async () => {
    render(<Articles category="emailed" />);
    await expect(screen.getByText(/Loading.../)).toBeInTheDocument();
  });

  it('matches snapshot', async () => {
    const { container } = render(<Articles category="emailed" />);
    await expect(container).toMatchSnapshot();
  });
});



describe('BookReview Component', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', async () => {
    render(<BookReview />);
    expect(screen.getByText(/Book Reviews/)).toBeInTheDocument();
  });

  it('displays search results when API request succeeds', async () => {
    const mockedBookData = {
      book_title: 'Test Book',
      book_author: 'Test Author',
      publication_dt: '2022-01-01',
      summary: 'Test summary',
    };

    const mockApiResponse = {
      results: [mockedBookData],
    };

    fetch.mockResolvedValueOnce({
      json: jest.fn().mockResolvedValueOnce(mockApiResponse),
    });

    render(<BookReview />);
    await screen.findByText(/Book Reviews/);
  });

  it('handles search input change and button click', async () => {
    const mockedBookData = {
      book_title: 'Test Book',
      book_author: 'Test Author',
      publication_dt: '2022-01-01',
      summary: 'Test summary',
    };

    const mockApiResponse = {
      results: [mockedBookData],
    };

    fetch.mockResolvedValueOnce({
      json: jest.fn().mockResolvedValueOnce(mockApiResponse),
    });

    render(<BookReview />);
    
    const inputElement = screen.getByPlaceholderText('Enter book title');
    const buttonElement = screen.getByText('Search');

    // Simulate user input
    fireEvent.change(inputElement, { target: { value: 'Test' } });
    expect(inputElement.value).toBe('Test');

    // Simulate button click
    fireEvent.click(buttonElement);

    await screen.findByText(/Book Reviews/);
  });
});

