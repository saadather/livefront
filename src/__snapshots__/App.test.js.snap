// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <nav
      class="left-nav"
    >
      <a
        class=""
        href="/most-popular"
      >
        Most Popular
      </a>
      <a
        class=""
        href="/book-review"
      >
        Book Review
      </a>
    </nav>
    <div
      class="main-content"
    />
  </div>
</div>
`;

exports[`Articles Component matches snapshot 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
